syz-cluster/pkg/app/secret.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2025 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package app
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+)
+
+type SecretKey string
+
+const (
+	SecretSMTPHost     SecretKey = "smtp_host"
+	SecretSMTPPort     SecretKey = "smtp_port"
+	SecretSMTPUser     SecretKey = "smtp_user"
+	SecretSMTPPassword SecretKey = "smtp_password"
+)
+
+type SecretManager interface {
+	Get(context.Context, SecretKey) (string, error)
+}
+
+// GCPSecretManager lazily queries and caches the secret values.
```

Comment:
I'm not sure gcp secrets caching is a good security practice... Didn't check though...

Reply:
They don't share any details re. this in the best practices doc: https://cloud.google.com/secret-manager/docs/best-practices.

They just warn against keeping secrets in env variables / making them accessible from a filesystem, but we don't do either of that.

Reply:
https://cheatsheetseries.owasp.org/cheatsheets/Secrets_Management_Cheat_Sheet.html#25-handling-secrets-in-memory

The idea - less time your secret lives in memory means better security.

Reply:
Rewrote the code to query credentials each time.